### PR TITLE
python3: packaging of new python3 plugins

### PR DESCRIPTION
**Pull request by pstorz**

### CODE REVIEW

File: core/platforms/packaging/bareos.spec [hunk 1]
```diff
@@ -524,11 +526,37 @@ Requires:   %{name}-common = %{version}
 Summary:    Python plugin for Bareos Director daemon
 Group:      Productivity/Archiving/Backup
 Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
+
+%package    director-python3-plugin
+Summary:    Python plugin for Bareos Director daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:535]: If you want the python-based-plugins to depend on the python-c-plugin, you will need a common virtual capability provided by both `*-python-plugin` and `*-python3-plugin` packages each and require that capability in the plugin packages.

However, historically `yum`'s depsolver (I'm not sure if `dnf` is the same) always selects the package with the shortest name when selecting the package to fulfil a dependency. This had been introduced to make sure we do not install a `*-debuginfo` package that would fulfill the dependency.
With the current naming scheme (i.e. `python` vs. `python3` you will most likely end up with the python2 plugin installed (at least on RHEL 7).

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #   02110-1301, USA.
 message("Entering ${CMAKE_CURRENT_SOURCE_DIR}")
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.12)
```
Comment (arogge) [inline CMakeLists.txt:21]: shouldn't we reflect this change in the RPM BuildRequires and the Debian pendant, too?

**Review by arogge — CHANGES_REQUESTED** [review]
I think we really need to give the python dependencies a thought. Do we want to default to python3? How can we achieve this?
Find my other comments inline.

File: docs/pkglists/CentOS_6.x86_64 [hunk 3]
```diff
@@ -13,19 +13,19 @@ x86_64/bareos-database-sqlite3
 x86_64/bareos-database-tools
 x86_64/bareos-devel
 x86_64/bareos-director
-x86_64/bareos-director-python-plugin
+x86_64/bareos-director-python2-plugin
 x86_64/bareos-filedaemon
```
Comment (arogge) [inline docs/pkglists/CentOS_6.x86_64:17]: are the common packages missing? i.e. `bareos-director-python-plugins-common` from the build?

File: docs/pkglists/Debian_8.0.i586 [hunk 4]
```diff
@@ -13,16 +13,16 @@ i386/bareos-database-tools
 i386/bareos-dbg
 i386/bareos-devel
 i386/bareos-director
-i386/bareos-director-python-plugin
+i386/bareos-director-python2-plugin
 i386/bareos-filedaemon
```
Comment (arogge) [inline docs/pkglists/Debian_8.0.i586:17]: is the common package missing here?

File: docs/pkglists/Debian_8.0.x86_64 [hunk 5]
```diff
@@ -13,16 +13,16 @@ amd64/bareos-database-tools
 amd64/bareos-dbg
 amd64/bareos-devel
 amd64/bareos-director
-amd64/bareos-director-python-plugin
+amd64/bareos-director-python2-plugin
 amd64/bareos-filedaemon
```
Comment (arogge) [inline docs/pkglists/Debian_8.0.x86_64:17]: is the common package missing here?

File: docs/pkglists/Fedora_29.x86_64 [hunk 6]
```diff
@@ -8,23 +8,23 @@ debug/x86_64/bareos-database-tools-debuginfo
 debug/x86_64/bareos-debuginfo
```
Comment (arogge) [inline docs/pkglists/Fedora_29.x86_64:8]: I belive we're not building Fedora 29 anymore. So maybe we could remove pkglist?

**Review by arogge — COMMENTED** [review]
Look good. I'd opt for some manual testing. Also I think there are pkglists that are not correct. Probably mostly for systems we no longer build for (e.g. CentOS/RHEL 6, Debian 8, etc.). It is probably best to remove these.

**Comment by pstorz** [comment]
Good point, removed the pkglists of OS we do not build anymore.

File: core/platforms/packaging/bareos.spec [hunk 7]
```diff
@@ -520,15 +522,49 @@ Group:      Development/Languages/C and C++
...
 
 %if 0%{?python_plugins}
-%package    director-python-plugin
+%package    director-python2-plugin
 Summary:    Python plugin for Bareos Director daemon
 Group:      Productivity/Archiving/Backup
 Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
+Provides:   director-python-plugin
+Obsoletes:  director-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:531]: shoud be "bareos-director-python-plugin"

Comment (arogge) [inline core/platforms/packaging/bareos.spec:531]:
```suggestion
Provides:   bareos-director-python-plugin
Obsoletes:  bareos-director-python-plugin
```

File: core/platforms/packaging/bareos.spec [hunk 8]
```diff
@@ -520,15 +522,49 @@ Group:      Development/Languages/C and C++
...
 
-%package    filedaemon-python-plugin
+%package    director-python3-plugin
+Summary:    Python plugin for Bareos Director daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
+Provides:   director-python-plugin
+Obsoletes:  director-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:539]:
```suggestion
Provides:   bareos-director-python-plugin
Obsoletes:  bareos-director-python-plugin
```

File: core/platforms/packaging/bareos.spec [hunk 9]
```diff
@@ -520,15 +522,49 @@ Group:      Development/Languages/C and C++
...
+Provides:   filedaemon-python-plugin
+Obsoletes:  filedaemon-python-plugin
+
+%package    filedaemon-python3-plugin
 Summary:    Python plugin for Bareos File daemon
 Group:      Productivity/Archiving/Backup
 Requires:   bareos-filedaemon = %{version}
+Requires:   bareos-filedaemon-python-plugins-common = %{version}
+Provides:   filedaemon-python-plugin
+Obsoletes:  filedaemon-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:561]:
```suggestion
Provides:   bareos-filedaemon-python-plugin
Obsoletes:  bareos-filedaemon-python-plugin
```

File: core/platforms/packaging/bareos.spec [hunk 10]
```diff
@@ -563,7 +599,23 @@ Requires:   bareos-filedaemon = %{version}
...
+Provides:   storage-python-plugin
+Obsoletes:  storage-python-plugin
+
+%package    storage-python3-plugin
+Summary:    Python plugin for Bareos Storage daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-storage = %{version}
+Requires:   bareos-storage-python-plugins-common = %{version}
+Provides:   storage-python-plugin
+Obsoletes:  storage-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:616]:
```suggestion
Provides:   bareos-storage-python-plugin
Obsoletes:  bareos-storage-python-plugin
```

File: core/platforms/packaging/bareos.spec [hunk 11]
```diff
@@ -520,15 +522,49 @@ Group:      Development/Languages/C and C++
...
+Requires:   bareos-director = %{version}
+
+
+%package    filedaemon-python2-plugin
+Summary:    Python plugin for Bareos File daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-filedaemon = %{version}
+Requires:   bareos-filedaemon-python-plugins-common = %{version}
+Provides:   filedaemon-python-plugin
+Obsoletes:  filedaemon-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:553]:
```suggestion
Provides:   bareos-filedaemon-python-plugin
Obsoletes:  bareos-filedaemon-python-plugin
```

File: core/platforms/packaging/bareos.spec [hunk 12]
```diff
@@ -563,7 +599,23 @@ Requires:   bareos-filedaemon = %{version}
...
 #Requires:   python-percona
 
-%package    storage-python-plugin
+%package    storage-python2-plugin
+Summary:    Python plugin for Bareos Storage daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-storage = %{version}
+Requires:   bareos-storage-python-plugins-common = %{version}
+Provides:   storage-python-plugin
+Obsoletes:  storage-python-plugin
```
Comment (arogge) [inline core/platforms/packaging/bareos.spec:608]:
```suggestion
Provides:   bareos-storage-python-plugin
Obsoletes:  bareos-storage-python-plugin
```

**Review by arogge — CHANGES_REQUESTED** [review]
Installation on RHEL 8 doesn't work.
Provided name doesn't contain "bareos-" Prefix.

**Comment by arogge** [comment]
I tested the installation on Debian 10. The python2 plugin package is preferred there. This is probably true for all other Debian-based distro's, too.

**Pull request merged**